proposals/open-telemetry/tracing.md

```diff
@@ -0,0 +1,263 @@
+    </dependency>
+    <dependency>
+      <groupId>io.opentelemetry</groupId>
+      <artifactId>opentelemetry-sdk</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.opentelemetry</groupId>
+      <artifactId>opentelemetry-exporter-jaeger</artifactId>
+    </dependency>
+</dependencies>
+```
+
+### Extension methods
+
+The `CqlSession` builder will support an extension method `withOpenTelemetry`, which will take an `OpenTelemetry` instance as a parameter.
+This `OpenTelemetry` instance will contain configuration for the resource and the exporter. The following is an example of the configuration:
+
+```java
+static OpenTelemetry initOpenTelemetry() {
+
+    ManagedChannel jaegerChannel =
+            ManagedChannelBuilder.forAddress("localhost", 14250).usePlaintext().build();
+
+    JaegerGrpcSpanExporter jaegerExporter =
+            JaegerGrpcSpanExporter.builder()
+                    .setChannel(jaegerChannel)
+                    .setTimeout(30, TimeUnit.SECONDS)
+                    .build();
+
+    Resource serviceNameResource =
+            Resource.create(Attributes.of(ResourceAttributes.SERVICE_NAME, "Demo App"));
+
+    SdkTracerProvider tracerProvider =
+            SdkTracerProvider.builder()
+                    .addSpanProcessor(SimpleSpanProcessor.create(jaegerExporter))
+                    .setResource(Resource.getDefault().merge(serviceNameResource))
+                    .build();
+    OpenTelemetrySdk openTelemetry =
+            OpenTelemetrySdk.builder().setTracerProvider(tracerProvider).build();
+
+    return openTelemetry;
+}
+```
+
+The `CqlSession` built with this method will have `OtelRequestTracker` registered as a request tracker. `OtelRequestTracker` will implement the `RequestTracker` interface and will be responsible for creating spans for each request.
+
+After [PR-1949](https://github.com/apache/cassandra-java-driver/pull/1949) is merged, the `RequestTracker` interface will include:
+```java
+void onSessionReady(@NonNull Session session) {}
+        
+void onRequestCreated(
+    @NonNull Request request,
+    @NonNull DriverExecutionProfile executionProfile,
+    @NonNull String requestLogPrefix) {}
+
+void onRequestCreatedForNode(
+    @NonNull Request request,
+    @NonNull DriverExecutionProfile executionProfile,
+    @NonNull Node node,
+    @NonNull String requestLogPrefix) {}
+
+void onSuccess(
+    long latencyNanos, @NonNull ExecutionInfo executionInfo, @NonNull String requestLogPrefix) {}
+
+void onError(
+    long latencyNanos, @NonNull ExecutionInfo executionInfo, @NonNull String requestLogPrefix) {}
+
+void onNodeSuccess(
+    long latencyNanos, @NonNull ExecutionInfo executionInfo, @NonNull String requestLogPrefix) {}
+
+void onNodeError(
+    long latencyNanos, @NonNull ExecutionInfo executionInfo, @NonNull String requestLogPrefix) {}
+```
+
+`OtelRequestTracker` will utilize the above methods.
+
+`OtelRequestTracker` will initialize OpenTelemetry exporter on `onSessionReady`. 
```

Comment:
I think `RequestTracker` is missing `close()` method to release resources (thread pools, channels etc.).